src/app/helpers/signals/signals.ts

```diff
@@ -0,0 +1,22 @@
+import { isSignal, Signal } from "@angular/core";
+
+/**
+ * A type that can either be a direct value of type T or a Signal that
+ * produces a value of type T.
+ * This is useful for route guards and directives that want to work in static,
+ * zone.js, and signal-based components/environments.
+ */
+export type PotentiallySignal<T> = T | Signal<T>;
```

Comment:
I'd have chosen `SignalOr<T>`, maybe `SignalOrScalar<T>`.

IDM `PotentialSignal` , though I do think the adverb is a bit much